lib/iris/tests/unit/lazy_data/test_as_lazy_data.py

```diff
@@ -42,6 +44,26 @@ def test_non_default_chunks(self):
         (result,) = np.unique(lazy_data.chunks)
         self.assertEqual(result, 24)
 
+    def test_dask_chunking(self):
+        data = np.arange(24)
+        chunks = (12,)
+        optimum = self.patch("iris._lazy_data._optimum_chunksize")
+        optimum.return_value = chunks
+        as_lazy_data(data, chunks=None, dask_chunking=True)
+        self.assertFalse(optimum.called)
+
+    def test_dask_chunking_error(self):
+        data = np.arange(24)
+        chunks = (12,)
+        optimum = self.patch("iris._lazy_data._optimum_chunksize")
+        optimum.return_value = chunks
+        with self.assertRaises(ValueError) as ar:
+            as_lazy_data(data, chunks=chunks, dask_chunking=True)
+        self.assertEqual(
+            str(ar.exception),
+            f"Dask chunking chosen, but chunks already assigned value {chunks}",
+        )
```

Comment:
Use [`assertRaisesRegex()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex) to check for the message at the same time.